lib/basketitem.rb

```diff
@@ -0,0 +1,35 @@
+class BasketItem
+  attr_reader :product
+  attr_accessor :quantity
+
+  WrongQuantityRequested = Class.new(StandardError)
+  NoSuchItem = Class.new(StandardError)
+
+  def initialize(product:, quantity: 1)
+    @product = product
+    @quantity = quantity
+
+    raise ArgumentError unless quantity.is_a?(Integer)
+    raise ArgumentError unless quantity > 0
+  end
+
+  def change_quantity(num)
+    raise ArgumentError unless num.is_a?(Integer)
+
+    new_quantity = @quantity + num
+
+    unless new_quantity < 0
+    ##(it's OK to set @quantity = 0 here, because Basket .summary method rejects such items)
+      @quantity = new_quantity
+    else
+      raise WrongQuantityRequested
+    end
```

Comment:
It looks weird that you have new variable that is assigned to the old one only when some condition is fulfilled. Maybe you can check that condition "in fly"?

Another thing is your comment. Why there are two hash characters on the beginning? Why there are brackets? Why there is no spacebar after hash? ...

imho `unless` should not be used with `else`. Use `if - else` with proper condition instead.